### PR TITLE
Add grpc-java v1.30.2 to interop test client matrix

### PR DESCRIPTION
**Pull request by sanjaypujare**

### CODE REVIEW

**Comment by voidzcy** [comment]
I don't have review permission in this repo.

**Comment by sanjaypujare** [comment]
@markdroth @srini100 @ejona86 can one of you approve and merge this PR? Thanks

File: tools/interop_matrix/client_matrix.py [hunk 1]
```diff
@@ -221,6 +221,7 @@ def __init__(self, patch=[], runtimes=[], testcases_file=None):
             ('v1.29.0', ReleaseInfo()),
             ('v1.30.0', ReleaseInfo()),
             ('v1.30.1', ReleaseInfo()),
+            ('v1.30.2', ReleaseInfo()),
```
Comment (ejona86) [inline tools/interop_matrix/client_matrix.py:224]: This should replace v1.30.1, and it seems we accidentally didn't replace v1.30.0.

Comment (sanjaypujare) [reply tools/interop_matrix/client_matrix.py:224]: Oh, okay. I will remove both 1.30.2 and 1.30.1 and replace 1.30.0 with 1.30.2.

**Comment by sanjaypujare** [comment]
I am going to merge this PR unless people have objections. Pls LMK

**Pull request merged**